### DISCUSSION TITLE
how to configure VSOMEIP3 for unicast communication

### DISCUSSION THREAD

**Discussion by fu9958** [COVESA/vsomeip#541] [Q&A] [answered]
Now,I have configured vsomeip3 for multicast communication yet,but I don't know how to configure VSOMEIP3 for unicast communication.
Anybody can give me an example json file?

**Comment by nachtfuchs** [comment]
What do you mean, when you say multicast communication and unicast communication? Service Discovery uses multicast to do its magic. Afterwards, the nodes can use their unicast addresses to communicate. Can you highlight your challenge in more detail?

**Reply by fu9958** [reply]
Sorry, maybe I didn't make it clear what my question was.
I just want to receive the offer message sent from server 10.1.0.10, which can only be received on IP 10.1.0.100. How should I configure JSON files? Thanks.

**Reply by nachtfuchs** [reply]
No problem. I know how hard it is to answer a good question.
Still not clear to me. There is a service discovery first, either in "automated" mode or with static IP addresses. Afterwards, a unicast communication can be established between the applications that use unicast communication. 
You said, that your offer message can only be received by one client. This is new to me. I did not know that this is possible. I thought that service discovery, where the offer service is communicated always uses multicast.

**Reply by fu9958** [reply]
Thank you all the same.
I'm not sure if this is possible too,  I just don't want that anybody would receive an offer message when their computer be connected to the CCU.Maybe I need to modify some code of vsomeip.

**Comment by fcmonteiro** [comment, marked as answer]
Currently what you describe is not possible to do.
You would need to create a "unicast target" in the SD configuration and use that instead of the multicast.
This is not implemented, but of course you can modify the vsomeip code.
